### PR TITLE
Add advanced parameters to geserver installer

### LINKED ISSUE #171: Add advanced parameters to geserver installer
The geserver installer originally had a few additional features planned that didn't make it into the first install script that was checked in. It should add:

1. PUBLISH_ROOT_VOLUME size checking
1. ASSET_ROOT and PUBLISH_ROOT volume checking
1. Some script parameters - parsing and execution (based on those provided by the Fusion installer)
1. Prompts and summary output similar Fusion installer

### PR DESCRIPTION
**Pull request by tnbezue**
Fixes #171

PUBLISH_ROOT_VOLUME size checking.
-- Checks if available space is gt 1048576 kb (is this a reasonable value)
-- Determine size and mount point for the publish root directory. Mount point so that user can ensure that proper volume is mounted.
This information is displayed in modification 4 (see below).
WIP 2) ASSET_ROOT and PUBLISH_ROOT volume checking
-- ASSET_ROOT volume checking is performed in install_fusion.sh (should it be duplicated in install_server.sh)
Other than size, are there addition checks that should be performed.

Some script parameters - parsing and execution (based on those provided by the Fusion installer)
-- Added options similar to Fusion installer:
-au -- Apache user option
-pgu -- Postgres user
-g -- Specifies group
-nobk -- Specifies not to execute backup
If a previous server installation is detected, user/group information is gathered from owners of specific files/directories associated with the function (i.e, GEAPACHEUSER_NAME is owner $PUBLISHER_ROOT/search_space)
If server not installed but a fusion installation detected, then group and fusion user info collected from systemrc file created by fusion installer.
If option is given, it can't specify a different user/group than used previous installation.
If new install, then defaults from previous version of script are used.
4) Prompts and summary output similar Fusion installer.

Created a summary output and prompt similar to Fusion installer

### CODE REVIEW

**Comment by tnbezue** [comment]
I updated docs/geedocs/5.3.0/answer/7160007.html in
171_advanced_params_geserver_installer branch.  It now shows as one of the
files changed in the pull request.

Terry

File: earth_enterprise/src/installer/install_server.sh [hunk 1]
```diff
@@ -25,6 +25,7 @@ SCRIPTDIR=`dirname $0`
 . $SCRIPTDIR/common.sh
 
 # config values
+ASSET_ROOT="/gevol/assets"
```
Comment (tst-rwildes) [inline earth_enterprise/src/installer/install_server.sh]: Server install doesn't need asset root.  Server can be installed without installing fusion.

File: earth_enterprise/src/installer/install_server.sh [hunk 2]
```diff
@@ -53,6 +45,20 @@ PRODUCT_NAME="$GEE"
 START_SERVER_DAEMON=1
 PROMPT_FOR_START="n"
 
+# user names
+DEFAULTGROUPNAME="gegroup"
+GRPNAME=$DEFAULTGROUPNAME
+GEAPACHEUSER_NAME="geapacheuser"
+GEPGUSER_NAME="gepguser"
+DEFAULTGEFUSIONUSER_NAME="gefusionuser"
```
Comment (tst-rwildes) [inline earth_enterprise/src/installer/install_server.sh]: These two lines don't need to be in server install as fusion is not required for server

File: earth_enterprise/src/installer/install_server.sh [hunk 3]
```diff
@@ -629,6 +748,81 @@ check_server_running()
...
+
+  if ! prompt_to_quit "X (Exit) the installer and cancel the installation - C (Continue) to install/upgrade."; then
+    return 1
+  else
+        echo -e "\nProceeding with installation..."
+    return 0
+    fi
+}
+
+load_systemrc_config()
```
Comment (tst-rwildes) [inline earth_enterprise/src/installer/install_server.sh]: Is this needed?

File: earth_enterprise/src/installer/install_server.sh [hunk 4]
```diff
@@ -629,6 +748,81 @@ check_server_running()
...
+  fi
+}
+
+# Get user/group info from previous installation.  Uses user/group for particular files/directories
+get_server_installation_info()
+{
+  GEAPACHEUSER_NAME=$(get_default_user "$PUBLISHER_ROOT/search_space" $GEAPACHEUSER_NAME)
+  GRPNAME=$(get_default_group "$PUBLISHER_ROOT/search_space" $GRPNAME)
+  GEPGUSER_NAME=$(get_default_user "$PGSQL_DATA" $GEPGUSER_NAME)
+  GEFUSIONUSER_NAME=$(get_default_user "$ASSET_ROOT" $GEFUSIONUSER_NAME)
```
Comment (tst-rwildes) [inline earth_enterprise/src/installer/install_server.sh]: Fusion might not be installed on the same machine as server.  So the server install shouldn't need this.

**Comment by tnbezue** [comment]
Removed references to ASSET_ROOT and load_systemrc_config

The lines referencing GEFUSIONUSER_NAME are included because, in the
modify_files function, $GEFUSIONUSER_NAME is saved to /etc/init.d/gevars.sh.


I'm available from 10 AM until 4 PM on Wednesday.  Can we do a quick (10 or
15 minute) meeting within that time window?

Terry

**Comment by tst-rwildes** [comment]
@tnbezue that is a good point about `gevars.sh` I forgot about that.  Code changes all look good and I tested things out on a CentOs 7 setup.  I did both new install and upgrade.  I'm going to give a bit more time for others to have the opportunity to review things before I merge the changes but if no one objects I'll be merging the PR this afternoon.

**Pull request merged**